lib/mongodb.js

```diff
@@ -348,9 +348,10 @@ MongoDB.prototype.getTypes = function() {
   return ['db', 'nosql', 'mongodb'];
 };
 
-MongoDB.prototype.getDefaultIdType = function() {
-  return ObjectID;
-};
+// SPIKE: get rid of this. TODO: remove other redundant code too
+// MongoDB.prototype.getDefaultIdType = function() {
+//   return ObjectID;
+// };
```

Comment:
I think this change will break compatibility with LB 3.x. Which is fine, we just need to clearly document it and remove `juggler@3` from the test suite.

As part of the doc changes, we should add a "Long Term Support" section to README (see [`strongloop/loopback`](https://github.com/strongloop/loopback#module-long-term-support-policy) for inspiration) and also explain which connector versions are compatible with which LoopBack versions.

Reply:
@hacksparrow can you please capture my comment above in the spike document?